services/socket.js

```diff
@@ -1,18 +1,24 @@
 const rollResult = require('../helper/rollResult');
 const calculateResult = require('../helper/calculateResult');
-const { TIME_LEFT_DEFAULT, PENDING_COUNTDOWN_TIME, COUNTDOWN_INTERVAL_SECOND } = require('../config');
+const { SECRET_KEY, TIME_LEFT_DEFAULT, PENDING_COUNTDOWN_TIME, COUNTDOWN_INTERVAL_SECOND } = require('../config');
+const jwt = require('jsonwebtoken');
 
 const activateSocket = (io) => {
   let timeLeft = TIME_LEFT_DEFAULT;
   let timeInterval;
   let userConnectedCounter = 0;
+  let userId;
 
-  const countdown = () => {
+  const countdown = async () => {
     if (timeLeft <= 0) {
       clearInterval(timeInterval);
       io.emit('processing-result');
       // Server roll result
-      rollResult();
+      await rollResult();
+
+      // Return result to player
+      const returnResult = await calculateResult(userId);
```

Comment:
```returnResult``` can be undefined, so you still emit socket even if ```returnResult = undefined```? I think when this case hapend you have to return error for client.